File: pregunta 2/preg2.scala
import scala.io.StdIn.readLine
object HolaMundo {
	
 	val suma:(Int,Int)=>Int = (x,y)=>x+y
 	val resta:(Int,Int)=>Int = (x,y)=>x-y
 	val multiplicacion:(Int,Int)=>Int = (x,y)=>x*y
 	val division:(Int,Int)=>Int = (x,y)=>x/y
 	
 	def main(args: Array[String]) = {
 		var op= 0 
 		while (op != -1){
 			println("1.- Suma\n2.- Resta\n3.- Multiplicacion\n4.- Division\n>\n")
 			op = readLine().toInt
 			println("Numero 1:>\n")
 			var x = readLine().toInt
 			println("Numero 2:>\n")
 			var y = readLine().toInt
 			if (op == 1){
 				println ("Resultado: "+ suma(x,y) );		
 			}else if (op ==2 ){
 				
 				println ("Resultado: "+ resta(x,y));		
 			}else if (op ==3 ){
 				
 				println ("Resultado: "+ multiplicacion(x,y));		
 			}else if (op ==4 ){
 				
 				println ("Resultado: "+ division(x,y));		
 			}else{
 				println ("Resultado: NA");		
 			}
 		}
 	}
}

File: pregunta 2/preg2.py
# -*- coding: utf-8 -*-
"""
Created on Mon Nov 29 22:26:57 2021

@author: Miguel
"""
suma = lambda x,y: x+y
resta = lambda x,y: x-y
multiplicacion = lambda x,y: x*y
division = lambda x,y: x/y

op = 0
while (op != -1):
    
    print ("1.- Sumar\n2.- Restar\n3.- Multiplicar\n4.- Dividir>")
    op = int(input())
    print ("Numero 1:")
    x = int(input())
    print ("Numero 1:")
    y = int(input())
    if (op == 1 ):
        print ("resultado: ", suma(x, y))
    else: 
        if (op == 2):
            print ("resultado: ",resta(x, y))
        else: 
            if (op == 3):
                print ("resultado: ",multiplicacion(x, y))
            else: 
                if (op == 4):
                    print ("resultado: ",division(x, y))
                else : print("NA") 